### PR TITLE
Document Agent/Fleet Server environment variables

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

- [x] Documents the environment variables found [here](https://github.com/elastic/beats/blob/23efd31aeaa6143a419c61c633f8c21d8a599264/x-pack/elastic-agent/pkg/agent/cmd/container.go#L59-L115).
- [x] Cleans up inconsistencies in the Kubernetes documentation
- [x] Adds env details to the Kubernetes documentation

### Related

Closes https://github.com/elastic/observability-docs/issues/623.
Related to https://github.com/elastic/observability-docs/pull/645.
Related to https://github.com/elastic/observability-docs/issues/589.
Related to https://github.com/elastic/observability-docs/issues/622.

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc [hunk 1]
```diff
@@ -63,14 +63,37 @@ Set the Elasticsearch settings before deploying the manifest:
   value: "https://somesuperhostiduuid.europe-west1.gcp.cloud.es.io:443"
 ------------------------------------------------
 
+// Begin collapsed section
+[%collapsible]
+.Configuration details
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc:68]: The configuration names in the example don't match the reference info: `ES_USERNAME` vs `ELASTICSEARCH_USERNAME`

Also might be worth stating what takes precedence: the settings in the manifest or the env vars.

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+// end::kibana-fleet-password[]
+
+// =============================================================================
+
+// tag::kibana-fleet-ca[]
+|
+[id="env-{type}-kibana-fleet-ca"]
+`KIBANA_FLEET_CA`
+
+| (string) The path to the certificate authority used to communicate with {kib}.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:84]: Should this be past tense?

**Review by dedemorton — COMMENTED** [review]
Someone on the dev team will need to review for technical accuracy, but the writing looks great! Just a few minor comments and questions.

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:84]: Good point. Fixed.

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc:68]: Oh wow, I totally missed that. I'll need someone from engineering to follow-up with on this.

File: docs/en/ingest-management/elastic-agent/configuration/env/container-envs.asciidoc [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+include::shared-env.asciidoc[tag=kibana-fleet-password]
+
+include::shared-env.asciidoc[tag=kibana-fleet-ca]
+
+|===
+
+[discrete]
+[[env-bootstrap-fleet-server]]
+== Bootstrap {fleet-server}
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/en/ingest-management/elastic-agent/configuration/env/container-envs.asciidoc:80]: There's `FLEET_ENROLLMENT_TOKEN` which is listed below.

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc [hunk 4]
```diff
@@ -63,14 +63,37 @@ Set the Elasticsearch settings before deploying the manifest:
...
-=== Running {agent} on master nodes
+=== Run {agent} on master nodes
 
-Kubernetes master nodes use https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/[taints]
+Kubernetes master nodes can use https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/[taints]
 to limit the workloads that can run on them. The manifest for standalone {agent} defines
-tolerations to run on master nodes, which collects metrics from the control plane
-components of Kuberentes (scheduler, controller manager).
-To disable {agent} from running on master nodes remove the following part of the Daemonset spec:
+tolerations to run on master nodes. These tolerations collect metrics from the control plane
```
Comment (ChrsMark) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc:94]:
```suggestion
tolerations to run on master nodes. Agents running on master nodes collect metrics from the control plane
```

Tolerations are used to make it possible for Agents' Pods to run on Master nodes.

**Review by ChrsMark — COMMENTED** [review]
K8s part looks good to me apart from one comment I left.

File: docs/en/ingest-management/elastic-agent/configuration/env/container-envs.asciidoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (dedemorton) [reply docs/en/ingest-management/elastic-agent/configuration/env/container-envs.asciidoc:80]: I'm thinking about the token that's used to enroll an agent in the Fleet Server policy (the equivalent to the `fleet-server-service-token` flag in this command).

```
sudo ./elastic-agent install -f --fleet-server-es=http://localhost:9200 \
--fleet-server-service-token=AAEAAWVsYXN0aWMvZmxlZXQtc2VydmVyL3Rva2VuLTE2MTkxMzg3MzIzMTg6dzEta0JDTmZUZGlDTjlwRmNVTjNVQQ
```

**Comment by bmorelli25** [comment]
Thanks, Chris!

@blakerouse, can you take a look at the environment variable part of this PR? There are some open review questions here that I'm hoping you'll be able to help with as well (or point me to someone who can). 🙇

File: docs/en/ingest-management/elastic-agent/configuration/env/container-envs.asciidoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (blakerouse) [reply docs/en/ingest-management/elastic-agent/configuration/env/container-envs.asciidoc:80]: Yes there is `FLEET_SERVER_SERVICE_TOKEN`.

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+
+// tag::kibana-fleet-setup[]
+|
+[id="env-{type}-kibana-fleet-setup"]
+`KIBANA_FLEET_SETUP`
+
+| (int) Set to `1` to enable {fleet} setup.
+Enabling {fleet} is required before {fleet-server} will start.
+
+*Default:* none
```
[the remark this answers is not in the document]
Comment (blakerouse) [reply docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:28]: The plugin is on by default but initialization of the Fleet plugin can only be performed by a superuser. This action in the container provides a way to bootstrap the Fleet plugin with the provided username/password, before the container proceeds. This ensures that Kibana is ready to go so Fleet Server can start up.

When this action is not used, then a user must manually login to Kibana and go to the Fleet page. Or an external process needs to perform the required API calls.

**Review by blakerouse — COMMENTED** [review]
Looks good to me, added 2 comments.

**Pull request merged**